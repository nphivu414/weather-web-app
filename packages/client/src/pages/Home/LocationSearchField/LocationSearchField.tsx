import * as React from 'react';
import { BiSearch } from 'react-icons/bi';

export type LocationSearchFieldProps = {
  hasSearchIcon?: boolean;
} & React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>;

const LocationSearchField: React.FC<LocationSearchFieldProps> = ({ hasSearchIcon = true, ...rest }) => {
  return (
    <>
      <div className="input-group mb-3">
        {hasSearchIcon && (
          <span className="input-group-text" id="basic-addon1">
            <BiSearch />
          </span>
        )}
        <input type="text" className="form-control" {...rest} />
      </div>
    </>
  );
};

export default LocationSearchField;
