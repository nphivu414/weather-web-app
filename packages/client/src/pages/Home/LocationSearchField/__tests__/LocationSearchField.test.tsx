import * as React from 'react';
import { shallow } from 'enzyme';
import { BiSearch } from 'react-icons/bi';

import LocationSearchField, { LocationSearchFieldProps } from '../LocationSearchField';

const props: LocationSearchFieldProps = {
  hasSearchIcon: false,
};

describe('LocationSearchField', () => {
  it('should match a snapshot', () => {
    expect(shallow(<LocationSearchField {...props} />)).toMatchSnapshot();
  });
  it('should show search icon', () => {
    const wrapper = shallow(<LocationSearchField {...props} hasSearchIcon />);
    expect(wrapper.find(BiSearch)).toHaveLength(1);
  });
});
