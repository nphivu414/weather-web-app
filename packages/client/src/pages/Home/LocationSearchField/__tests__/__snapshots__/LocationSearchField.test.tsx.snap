// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationSearchField should match a snapshot 1`] = `ShallowWrapper {}`;
